pkg/cluster/spec/alertmanager.go

```diff
@@ -129,12 +131,15 @@ func (i *AlertManagerInstance) InitConfig(e executor.Executor, clusterName, clus
 	}
 
 	// transfer config
-	fp = filepath.Join(paths.Cache, fmt.Sprintf("alertmanager_%s.yml", i.GetHost()))
-	if err := config.NewAlertManagerConfig().ConfigToFile(fp); err != nil {
-		return err
+	configPath := filepath.Join(paths.Cache, fmt.Sprintf("alertmanager_%s.yml", i.GetHost()))
+	if spec.ConfigFilePath != "" {
+		configPath = spec.ConfigFilePath
+	} else if err := config.NewAlertManagerConfig().ConfigToFile(configPath); err != nil {
+		return errors.Annotate(err, "failed to generate alertmanager config")
 	}
+
 	dst = filepath.Join(paths.Deploy, "conf", "alertmanager.yml")
-	return e.Transfer(fp, dst, false)
+	return i.TransferLocalConfigFile(e, configPath, dst)
 }
```

Comment:
```go
if spec.ConfigFilePath != "" {
    dst = filepath.Join(paths.Deploy, "conf", "alertmanager.yml")
    return i.TransferLocalConfigFile(e, spec.ConfigFilePa, dst)
}

configPath := filepath.Join(paths.Cache, fmt.Sprintf("alertmanager_%s.yml", i.GetHost()))
return config.NewAlertManagerConfig().ConfigToFile(configPath)
```
Keep the coding style consistent with other places.

Reply:
They have different logic structure.
It's
```golang
if xxx {
} else if xxx {
}
```
not
```golang
if xxx {
} else {
}
```

Reply:
I think the suggestion is suitable for this scenario.